src/libraries/System.Private.CoreLib/src/System/IO/FileStream.Windows.cs

```diff
@@ -389,6 +389,7 @@ private void SetLengthCore(long value)
                 SeekCore(_fileHandle, value, SeekOrigin.Begin);
             if (!Interop.Kernel32.SetEndOfFile(_fileHandle))
             {
+                SeekCore(_fileHandle, origPos, SeekOrigin.Begin);
                 int errorCode = Marshal.GetLastWin32Error();
```

Comment:
SeekCore may overwrite the last error. The last error from SetEndOfFile needs to be captured before calling SeekCore.

Reply:
Yup, thanks.

Another approach could be to wrap the `SetEndOfFile` call in a try-finally and restore `Position` in the finally block as suggested in https://github.com/dotnet/runtime/issues/8307 issue description, thoughts?